### ISSUE TITLE
RA8875 with Adafruit Display does not work

### ISSUE THREAD

**Issue by tobi807** [lovyan03/LovyanGFX#588]
**Carefully written Issues are more likely to be given priority.**
**丁寧に記述された報告は優先して対応される可能性が高くなります。**

## Environment ( 実行環境 )

- MCU or Board name:  [ESP-WROOM-32 on custom Board]
- Panel Driver IC:    [RA8875]
- Bus type:           [SPI]
- LovyanGFX version:  [v1.1.16]
- FrameWork version:   [ArduinoESP32]
- Build Environment:   [PlatformIO]
- Operating System:   [Windows]

### Expected Behavior ( 期待される動作 )
Show something, e.g. a circle or a rect. Later I would like to have it working with lvgl.

### Actual Behavior ( 実際の動作 )
display stays white and draws nothing at all. 

I use the Adafruit RA8875 together with an 5" Display and a ESP32 Module. The controller boots up than the display turns from black to white and thats it. 
I can manipulate the Brightness of the Backlight if I change line 55 in Panel_RA8875.cpp what indicates that the SPI Communication works.
```cpp
   0x8b      , 1, 0x20, //Backlight brightness setting //Brightness parameter 0xff-0x00
```
I tried multiple settings (trial and error really) and drawing methods (draw dirctly and with sprites) but nothing works. 
If I use the AdafruitGFX Library on the exact same setup it works perfectly fine.

If you need any additional information I'll give my best to provide them. I just dont know how to narrow down the problem

![aufbau](https://github.com/lovyan03/LovyanGFX/assets/175124362/8e046894-1b32-4bd2-88e4-441b8a68b85b)
The setup. The wires from the esp32 to the RA8875 Board run on the back side.

### Code to reproduce this issue ( 再現させるためのコード )
[main.cpp](https://gist.github.com/tobi807/b71466fd15161353016ed6a1a4848c59)
[platformio.ini](https://gist.github.com/tobi807/1c210e98d1758d74028c1f4ed1dda42c)
[SPI Communication](https://gist.github.com/tobi807/087f6ef5878f17dd7be39e6d67b325bc)

## additional information
[Adafruit Driver Board](https://www.adafruit.com/product/1590)
[circuit diagramm of driver board](https://cdn-learn.adafruit.com/assets/assets/000/124/420/original/lcds___displays_RA8875_Schematic.png?1694716073)
[Adafruit Display](https://www.adafruit.com/product/1596)

### SPI oscilloscope snippet
I analyzed the signals directly on the RA8875 driver board

Pico Scope file:
[lovyan_spi_comm.zip](https://github.com/user-attachments/files/16144004/lovyan_spi_comm.zip)

overview
![spi1](https://github.com/lovyan03/LovyanGFX/assets/175124362/eab25990-07da-4e43-9eee-422d705bba1d)
magnified
![spi2](https://github.com/lovyan03/LovyanGFX/assets/175124362/1534e603-ce5c-4042-9f47-46a9d6ed6359)

decoded
|Data MOSI  |Data MISO  |
|-----------|-----------|
|80 88      |FF FF      |
|00 0B      |FF FF      |
|80 89      |FF FF      |
|00 01      |FF FF      |
|80 04      |FF FF      |
|00 82      |FF FF      |
|80 15      |FF FF      |
|00 02      |FF FF      |
|80 16      |FF FF      |
|00 03      |FF FF      |
|80 17      |FF FF      |
|00 01      |FF FF      |
|80 18      |FF FF      |
|00 03      |FF FF      |
|80 1B      |FF FF      |
|00 0F      |FF FF      |
|80 1C      |FF FF      |
|00 00      |FF FF      |
|80 1D      |FF FF      |
|00 0E      |FF FF      |
|80 1E      |FF FF      |
|00 06      |FF FF      |
|80 1F      |FF FF      |
|00 01      |FF FF      |
|80 8A      |FF FF      |
|00 80      |FF FF      |
|80 8A      |FF FF      |
|00 81      |FF FF      |
|80 8B      |FF FF      |
|00 7F      |FF FF      |
|80 01      |FF FF      |
|00 80      |FF FF      |
|80 14 00 64|FF FF FF FF|
|...||

**Comment by tobi807** [comment]
**tl;dr** 
just add
```c
0xc7      , 1, 0x01,  // set Extra General Purpose IO Register to Output (used to enable Display by Adafruit)
```
to 
```c
const uint8_t* Panel_RA8875::getInitCommands(uint8_t listno) const
```
inside **Panel_RA8875.cpp** right above PWM Setting.

**long Version**
The issue trouble me and left me no rest so I kept investigating. Aaaand after a few more hours I found the Problem! And, also as expected, it was something trivial but very hard to find. Adafruit uses on their RA8875 Board the GPIOX to Enable/Disable the Display (tiny ON/OFF label on Pin 46 in the schematic). These GPIO must be set to Output, otherwise the Display wont get the Signal to turn on. So just one register write solved the Problem.

Regards,
Tobi
